### PR TITLE
[FIX] fields: read unstored one2many field without search function

### PR DESCRIPTION
**Pull request by abla001**
To Reproduce:
=============

- on a contact add a one2many field using studio
- in related field choose a field that is not stored and doesn't have a search function implemented
- close studio and notice all the lines of the selected field are listed on the contact even if are not linked to it

Problem:
========
- when searching a field that is not stored and doesn't have a search function, all the lines are returned

Solution:
=========
in this usecase filter the returned lines and only keep the ones linked to the record

opw-3265982

### CODE REVIEW

File: odoo/fields.py [hunk 1]
```diff
@@ -4321,15 +4321,21 @@ def read(self, records):
         domain = self.get_domain_list(records) + [(inverse, 'in', records.ids)]
         lines = comodel.search(domain)
 
+        # group lines by inverse field (without prefetching other fields)
```
Comment (nle-odoo) [inline odoo/fields.py:4324]: the comment should not be moved I think (the method is an utility, but the comment is about the line below)

**Review by nle-odoo — APPROVED** [review]
seems good to me

**Review by ryv-odoo — COMMENTED** [review]
Hello @abla001 , 
I don't understand how it is possible to have a one2many linked to an unsearchable inverse. How can [lines](https://github.com/odoo/odoo/pull/121664/files#diff-eb000d15355b38936d1b7f7506465ad081a3d96424c1f64dd1008ef3d4ac6427R4321-R4322) work then ?

Also, for this kind of fix, we want to cover it by at least one test to avoid any future regression.

**Comment by abla001** [comment]
> Hello @abla001 , I don't understand how it is possible to have a one2many linked to an unsearchable inverse. These [lines](https://github.com/odoo/odoo/pull/121664/files#diff-eb000d15355b38936d1b7f7506465ad081a3d96424c1f64dd1008ef3d4ac6427R4321-R4322) shouldn't work then ?
> 
> Also, for this kind of fix, we want to cover it by at least one test to avoid any future regression.

Hi @ryv-odoo, yes I agree those lines shouldn't work but they do, when we search o2m not stored and unsearchable we got all the lines, @rco-odoo explained to me that we should not have allowed that from the beginning, so I gave Studio team a feedback to not give this possibility, and a feedback to Stock team to add search method to that field.

For test you asked for, as this is wrong from the beginning what should I test exactly ? because if I replicate the usecase of the ticket (read the field and check to have only the linked lines ...) then we are testing a scenario that should not be there, can you guide me in this point please ?

**Comment by abla001** [comment]
@ryv-odoo yes exactly, that's why I created the feedbacks 
feedback for studio : https://www.odoo.com/web#cids=1&menu_id=4720&action=333&active_id=809&model=project.task&view_type=form&id=3330205

feedback for Stock to add search function to `last_delivery_partner` :
https://www.odoo.com/web#cids=1&menu_id=4720&action=333&active_id=809&model=project.task&view_type=form&id=3330221

but for creating a test in studio tests I think we can't do it before they implement it :disappointed:, in my opinion I don't see a good test for the moment

**Comment by ryv-odoo** [comment]
Nice job and Thanks @abla001 .
@robodoo r+

**Pull request closed without merging**